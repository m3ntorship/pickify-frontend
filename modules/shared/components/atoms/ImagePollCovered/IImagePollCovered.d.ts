declare namespace IImagePollCovered {
  export interface IProps extends IHandlers {
    isOneImageVote: boolean;
    like?: boolean;
    dislike?: boolean;
    id: string;
  }
  export interface IHandlers {
    onOptionClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  }
}

export { IImagePollCovered };
