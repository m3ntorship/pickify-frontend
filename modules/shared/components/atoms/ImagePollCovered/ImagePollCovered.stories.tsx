import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import ImagePollCovered from './ImagePollCovered';
import type { IImagePollCovered } from './IImagePollCovered';

export default {
  title: 'Atoms/ImagePollCovered',
  component: ImagePollCovered,
  argTypes: {
    isOneImageVote: {
      control: {
        type: 'boolean',
      },
    },
    like: {
      control: {
        type: 'boolean',
      },
    },
    dislike: {
      control: {
        type: 'boolean',
      },
    },
  },
};

const Template: Story<IImagePollCovered.IProps> = (args): ReactElement => (
  <ImagePollCovered {...args} />
);

export const ImagePollCoveredStory = Template.bind({});
ImagePollCoveredStory.args = {
  isOneImageVote: false,
  like: false,
  dislike: false,
};
