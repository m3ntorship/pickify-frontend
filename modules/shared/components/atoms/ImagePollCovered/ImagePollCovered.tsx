import type { ReactElement, FC } from 'react';
import React from 'react';
import type { IImagePollCovered } from './IImagePollCovered';
import FilledHeartIcon from '../../icons/filledHeart.svg';
import FilledDislikeIcon from '../../icons/filledDislike.svg';
import FilledLikeIcon from '../../icons/filledLike.svg';
import styles from './ImagePollCovered.module.css';

const ImagePollCovered: FC<IImagePollCovered.IProps> = ({
  isOneImageVote,
  like,
  dislike,
}): ReactElement => {
  return (
    <div className="flex justify-end">
      {!isOneImageVote && (
        <button type="button" className={`${styles.love} group`}>
          <FilledHeartIcon className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white" />
        </button>
      )}
      {isOneImageVote && (
        <div className="flex">
          {dislike && (
            <button type="button" className={`${styles.dislike} group`}>
              <FilledDislikeIcon className="fill-error stroke-current stroke-1 text-white group-hover:fill-white" />
            </button>
          )}
          {like && (
            <button type="button" className={`${styles.like} group`}>
              <FilledLikeIcon className="fill-primary stroke-current stroke-1 text-white group-hover:fill-white" />
            </button>
          )}
        </div>
      )}
    </div>
  );
};

export default ImagePollCovered;
