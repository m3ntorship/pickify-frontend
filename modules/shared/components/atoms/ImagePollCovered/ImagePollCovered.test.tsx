import React from 'react';
import renderer from 'react-test-renderer';
import ImagePollCovered from './ImagePollCovered';

describe('ImagePollCovered', () => {
  it('should render like button when passing isOneImageVote=[true] and like=[true]', () => {
    const tree = renderer
      .create(
        <ImagePollCovered
          isOneImageVote
          like
          id="option_1"
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render dislike button when passing isOneImageVote=[true] and dislike=[true]', () => {
    const tree = renderer
      .create(
        <ImagePollCovered
          isOneImageVote
          dislike
          id="option_1"
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render heart button when passing isOneImageVote=[false]', () => {
    const tree = renderer
      .create(
        <ImagePollCovered
          isOneImageVote={false}
          id="option_1"
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
