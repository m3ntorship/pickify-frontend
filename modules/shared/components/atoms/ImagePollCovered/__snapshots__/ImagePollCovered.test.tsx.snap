// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePollCovered should render dislike button when passing isOneImageVote=[true] and dislike=[true] 1`] = `
<div
  className="flex justify-end"
>
  <div
    className="flex"
  >
    <div
      className="group mr-4"
    >
      <button
        className="btn btn-like btn-large btn-rounded"
        disabled={false}
        id="option_1"
        onClick={[Function]}
        type="button"
      >
        <span>
          <svg
            className="fill-error stroke-current stroke-1 text-white group-hover:fill-white"
            data-file-name="SvgFilledDislike"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ImagePollCovered should render heart button when passing isOneImageVote=[false] 1`] = `
<div
  className="flex justify-end"
>
  <div
    className="group"
  >
    <button
      className="btn btn-like btn-large btn-rounded"
      disabled={false}
      id="option_1"
      onClick={[Function]}
      type="button"
    >
      <span>
        <svg
          className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
          data-file-name="SvgFilledHeart"
        />
      </span>
    </button>
  </div>
</div>
`;

exports[`ImagePollCovered should render like button when passing isOneImageVote=[true] and like=[true] 1`] = `
<div
  className="flex justify-end"
>
  <div
    className="flex"
  >
    <div
      className="group"
    >
      <button
        className="btn btn-like btn-large btn-rounded"
        disabled={false}
        id="option_1"
        onClick={[Function]}
        type="button"
      >
        <span>
          <svg
            className="fill-primary stroke-current stroke-1 text-white group-hover:fill-white"
            data-file-name="SvgFilledLike"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;
